client/src/main/java/io/pravega/client/security/auth/JwtTokenProviderImpl.java

```diff
@@ -0,0 +1,236 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.security.auth;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import io.pravega.client.stream.impl.Controller;
+import io.pravega.common.Exceptions;
+import io.pravega.common.concurrent.Futures;
+import io.pravega.common.util.ConfigurationOptionsExtractor;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.time.Instant;
+import java.util.Base64;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Provides JWT-based delegation tokens.
+ */
+@Slf4j
+public class JwtTokenProviderImpl implements DelegationTokenProvider {
+
+    /**
+     * Represents the threshold for triggering delegation token refresh.
+     */
+    @VisibleForTesting
+    static final int DEFAULT_REFRESH_THRESHOLD = 5;
```

Comment:
This name should be suffixed with a unit.

Reply:
Sure, I've added the unit of time as a suffix now.